### PR TITLE
Production server

### PR DESCRIPTION
**Pull request by ricwo**
**Proposed changes**:
- updates `rasa up` command

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa/cli/scaffold.py [hunk 1]
```diff
@@ -121,17 +139,22 @@ def run(args: argparse.Namespace) -> None:
           "bad ass contextual assistant, checkout our quickstart guide"
           "here: https://rasa.com/docs/core/quickstart \n\n"
           "Now let's start! 👇🏽\n")
-    path = questionary.text("Please enter a folder path where I should create "
-                            "the initial project [default: current directory]",
-                            default=".").ask()
+
+    if not args.no_prompt:
+        path = questionary.text(
```
Comment (tmbo) [inline rasa/cli/scaffold.py]: questionary `1.1.0` has a great new feature there 😉 

`path = questionary.text("hello?").skip_if(args.no_prompt, default=".").ask()`

File: rasa/cli/up.py [hunk 2]
```diff
@@ -1,80 +1,200 @@
...
+
+    configure_colored_logging(args.loglevel)
+    configure_file_logging(args.loglevel, args.log_file)
+
+    if args.production:
+        print_success("Starting Rasa X in production mode... 🚀")
+        _core_service(args)
+    else:
+        print_success("Starting Rasa X in local mode... 🚀")
+        try:
```
Comment (ricwo) [inline rasa/cli/up.py]: @tmbo can we get rid of this `try` then and use `is_rasa_x_installed`?

File: rasa/core/server.py [hunk 3]
```diff
@@ -623,13 +624,13 @@ def create_app(
                 evaluation = run_evaluation(data_path, model_path)
                 return response.json(evaluation)
             except ValueError as e:
-                return ErrorResponse(
+                raise ErrorResponse(
                     400,
                     "FailedIntentEvaluation",
-                    "Evaluation could not be created. Error: {}".format(e),
+                    "Evaluation could not be created. " "Error: {}".format(e),
```
Comment (tmbo) [inline rasa/core/server.py]:
```suggestion
                    "Evaluation could not be created. Error: {}".format(e),
```

File: rasa/core/tracker_store.py [hunk 4]
```diff
@@ -283,32 +287,96 @@ def __init__(
...
+        if url is not None:
+            engine_url = URL(
+                dialect, username, password, url, database=login_db if login_db else db
+            )
+        else:
+            engine_url = URL(
+                dialect,
+                username,
+                password,
+                host,
```
Comment (tmbo) [inline rasa/core/tracker_store.py]: why do we need to pass these in separately, can't we always use the url format?

File: rasa/core/tracker_store.py [hunk 5]
```diff
@@ -283,32 +287,96 @@ def __init__(
...
+                self.engine = create_engine(engine_url)
 
-        self.Base.metadata.create_all(self.engine)
+                if login_db:
+                    self._create_database_and_update_engine(db, engine_url)
+
+                try:
+                    self.Base.metadata.create_all(self.engine)
+                except (
+                    sqlalchemy.exc.OperationalError,
```
Comment (tmbo) [inline rasa/core/tracker_store.py]: what are these? please add an explanation to the `pass`

File: rasa/core/tracker_store.py [hunk 6]
```diff
@@ -283,32 +287,96 @@ def __init__(
...
 
+    def _create_database_and_update_engine(self, db, engine_url):
+        from sqlalchemy import create_engine
+
+        self._create_database(self.engine, db)
+        engine_url.database = db
+        self.engine = create_engine(engine_url)
+
+    @staticmethod
+    def _create_database(engine, db_name):
```
Comment (tmbo) [inline rasa/core/tracker_store.py]: please add a description (and types)

File: rasa/core/tracker_store.py [hunk 7]
```diff
@@ -283,32 +287,96 @@ def __init__(
...
+                sqlalchemy.exc.IntegrityError,
+            ) as e:
+                logger.warning(e)
+                sleep(5)
 
         logger.debug("Connection to SQL database '{}' successful".format(db))
 
         super(SQLTrackerStore, self).__init__(domain, event_broker)
 
+    def _create_database_and_update_engine(self, db, engine_url):
```
Comment (tmbo) [inline rasa/core/tracker_store.py]: please add a description (and types)

File: rasa/core/tracker_store.py [hunk 8]
```diff
@@ -283,32 +287,96 @@ def __init__(
...
+                "SELECT 1 FROM pg_catalog.pg_database "
+                "WHERE datname = '{}'".format(db_name)
+            )
+        )
+        exists = cursor.fetchone()
+        if not exists:
+            try:
+                cursor.execute("CREATE DATABASE {}".format(db_name))
+            except psycopg2.IntegrityError as e:
+                logger.debug("Could not create database '{}': {}" "".format(db_name, e))
```
Comment (tmbo) [inline rasa/core/tracker_store.py]: this sounds a bit more severe than a debug, no?

File: rasa/cli/scaffold.py [hunk 9]
```diff
@@ -15,16 +15,27 @@ def add_subparser(
...
-        "Do you want me to train an initial model for the bot? 💪🏽"
-    ).ask()
+    print_success("Finished creating project structure.")
+
+    if not args.no_prompt:
+        should_train = questionary.confirm(
+            "Do you want me to train an initial " "model for the bot? 💪🏽"
+        ).ask()
+    else:
+        print_success("Training an initial model...")
```
Comment (tmbo) [inline rasa/cli/scaffold.py]: I think we should move the print after `if should_train` and use `.skip_if` for the confirm without the surrounding `if no args.no_prompt`

File: rasa/cli/scaffold.py [hunk 10]
```diff
@@ -47,9 +58,12 @@ def print_run_or_instructions(args: argparse.Namespace, path: Text) -> None:
     from rasa.core import constants
     import questionary
 
-    should_run = questionary.confirm(
-        "Do you want to speak to the trained bot on the command line? 🤖"
-    ).ask()
+    if not args.no_prompt:
+        should_run = questionary.confirm(
+            "Do you want to speak to the trained bot " "on the command line? 🤖"
+        ).ask()
```
Comment (tmbo) [inline rasa/cli/scaffold.py]: `.skip_if(args.no_prompt, default=False)`

File: requirements.txt [hunk 11]
```diff
@@ -41,12 +41,14 @@ sanic==18.12.0
...
+questionary==1.1.0
 python-socketio==3.1.1
 pydot==1.4.1
 async_generator==1.10
 keras-applications==1.0.6
 keras-preprocessing==1.0.5
-sqlalchemy~=1.2.17
+SQLAlchemy~=1.2.18
+# --no-binary can be removed for version >= 2.8 http://initd.org/psycopg/articles/2018/02/08/psycopg-274-released/
+psycopg2==2.7.7 --no-binary psycopg2
```
Comment (tmbo) [inline requirements.txt]: is there a reason we are not using `2.8.2`?

**Pull request merged**